tests/integration/test_k8s_with_cluster.py

```diff
@@ -284,6 +284,32 @@ def test_workflow_will_cleanup_jobs_and_rollback_new_deployments_that_yield_erro
             delete_cluster_role_binding(workflow_sa_crb)
 
 
+def test_workflow_will_run_failure_stage_on_workflow_failure(
+    test_namespace: str, docker_image: str
+):
+    try:
+        process_one = run(
+            [
+                "bodywork",
+                "workflow",
+                f"--namespace={test_namespace}",
+                "https://github.com/bodywork-ml/bodywork-failing-test-project",  # noqa
+                "master",
+                f"--bodywork-docker-image={docker_image}",
+            ],
+            encoding="utf-8",
+            capture_output=True,
+        )
+
+        expected_output_0 = "ERROR - workflow_execution.run_workflow - failed to execute workflow for master branch"  # noqa
+        expected_output_1 = "successfully ran stage=on_fail_stage"
```

Comment:
You could print to stdout from the on-failure-stage and look for that message, as well, as an `expected_output_3`?

Reply:
Yeah I can easily add that in, it already is printing something to stdout.